linode/manager
# refactor: [M3-7277] - SAST Scan Findings: Path Traversal Vulnerability

## Description 📝
This PR resolve path traversal warning vulnerability in below files.

- [x] constants.mjs
- [x] deleteChangeset.mjs
- [x] generate-changelogs.mjs  
- [x] package-versions/index.js 

## Changes  🔄
List any change relevant to the reviewer.
- Implemented sanitizeFileName and safeJoinPath in generate-changelog.mjs
- Implemented sanitizeInput, validatePackage, validateFileName and safePathJoin in constants.mjs
- Implemented sanitizeFileName and safeJoinPath in deleteChangeset.mjs
- Implemented sanitizePackageName in deleteChangeset.mjs

## Target release date 🗓️
10/16

### Verification steps
(How to verify changes)
- No regression in generate changelogs, delete changelogs and package versions
- Verify the branch (develop vs M3-7277) SAST scan job results warnings for the above files got reduced from 33 to 29.
- Reach me incase of any further questions.

## As an Author I have considered 🤔

*Check all that apply*
